pkg/virt-launcher/virtwrap/manager.go

```diff
@@ -863,10 +867,25 @@ monitorLoop:
 			l.setMigrationResult(vmi, true, fmt.Sprintf("%v", err), "")
 			break monitorLoop
 		case libvirt.DOMAIN_JOB_CANCELLED:
-			logger.Info("Migration was canceled")
+			logger.Info("Migration job was canceled")
 			l.setMigrationResult(vmi, true, "Live migration aborted ", v1.MigrationAbortSucceeded)
 			break monitorLoop
 		}
+
+		select {
+		case passedErr := <-migrationErr:
```

Comment:
Can it be that this is still racy? What if we get a migrationErr immediately after `GetJobStats`? The job may report that it is still running, but we would immediately afterwards see a migrationErr. Can that be?

Reply:
I think it's very unlikely... 
Previously we had a 400ms gap between the last `getJobInfo` update and the error, while now it's close to 0.
I can try to further enhance this.

Reply:
Yes pl ase. It should be pretty easy to remove the race, by simply always determining the job state.